### PR TITLE
chore: added multi-root workspaces config for vscode

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-2649

Part of the 20% initiative!

In this PR, I added a [multi-root workspaces](https://code.visualstudio.com/docs/editor/multi-root-workspaces) configuration to the repo and I added an extension for unit testing some packages.

To test this PR, I highly recommend that you:
1. Checkout this branch
2. When prompted, install recommended extensions
3. Open the multi-root workspaces (can be done by running `code ui-kit.code-workspace` in the terminal).
4. Press CMD+P (macOS) or CTRL+P (Windows)
5. Search for "category-field-suggestions.test.ts"
6. Make any change to the file. Tests should auto-run.
7. Look at the "Testing" pane on the left-hand-side of vscode. You should see 4 projects and their tests.

I've been using a similar config (excluding the Jest part) for maybe about 2 months and it worked well.

### CODE REVIEW

File: ui-kit.code-workspace [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    },
+    {
+      "name": "⚙️ NX Adapter",
+      "path": "./utils/nx-plugin"
+    }
+  ],
+  "settings": {
+    "jest.disabledWorkspaceFolders": [
+      "✨ ui-kit",
+      "📦 Atomic",
```
Comment (btaillon) [inline ui-kit.code-workspace:83]: Atomic's tests aren't compatible with the Jest test runner unfortunately, and I haven't found a way to override settings for a single project.

File: ui-kit.code-workspace [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      "📦 Atomic for React",
+      "📦 Atomic for Angular",
+      "📄 Atomic Storybook documentation",
+      "📦 Atomic Hosted Page",
+      "🚀 IIFE samples",
+      "🚀 Stencil samples",
+      "🚀 React samples (Headless)",
```
Comment (btaillon) [inline ui-kit.code-workspace:90]: Same issue as Atomic tests.

File: ui-kit.code-workspace [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      "🚀 React samples (Atomic)",
+      "🚀 Next.js samples (Atomic)",
+      "🚀 Angular samples",
+      "🚀 Vue.js samples",
+      "⚙️ Release utilities",
+      "⚙️ NX Adapter",
+    ],
+    "jest.autoRevealOutput": "off",
```
Comment (btaillon) [inline ui-kit.code-workspace:98]: If this setting isn't turned off, then the Jest extension opens a bunch of terminals whenever you open the multi-root workspace.

File: ui-kit.code-workspace [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    "npm.packageManager": "npm",
+    "files.exclude": {
+      "**/node_modules/": true,
+      "packages/": true,
+      "utils/": true
+    },
+    "search.exclude": {
+      "dist/": true
+    },
```
Comment (btaillon) [inline ui-kit.code-workspace:107]: Been testing those for a while, they work well.

File: ui-kit.code-workspace [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    "[javascript]": {
+      "editor.formatOnSave": true
+    },
+    "[typescriptreact]": {
+      "editor.formatOnSave": true
+    },
+    "[postcss]": {
+      "editor.formatOnSave": true
+    },
```
Comment (btaillon) [inline ui-kit.code-workspace:121]: IMO, format on save is a really nice time saver. Got used to it thanks to Flutter.

File: .vscode/extensions.json [hunk 6]
```diff
@@ -3,6 +3,7 @@
     "dbaeumer.vscode-eslint",
     "esbenp.prettier-vscode",
     "redhat.vscode-yaml",
-    "streetsidesoftware.code-spell-checker"
+    "streetsidesoftware.code-spell-checker",
+    "orta.vscode-jest"
```
Comment (btaillon) [inline .vscode/extensions.json:7]: The only Jest extension that supports multi-root workspaces AFAIK, and each of them has very bad monorepo support.

File: ui-kit.code-workspace [hunk 7]
```diff
@@ -0,0 +1,123 @@
+{
+  "folders": [
+    {
+      "name": "✨ ui-kit",
```
Comment (mrrajamanickam-coveo) [inline ui-kit.code-workspace:4]: Is this to access files at the root level? Would it include all packages etc as well? Or could packages be excluded here for simplicity?

File: ui-kit.code-workspace [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+      "path": "./"
+    },
+    {
+      "name": "📦 Auth",
```
Comment (mrrajamanickam-coveo) [inline ui-kit.code-workspace:8]: Does this order reflect in VSCode UI as well? If yes then can we order by importance / frequency-of-editing .. e.g. headless, atomic, followed by integration packages e.g. atomic-react, followed by samples etc

**Review by mrrajamanickam-coveo — APPROVED** [review]
Looks good, if its opt-in and not enabled by default 👍

File: ui-kit.code-workspace (hunk 7 above)
Comment (btaillon) [reply ui-kit.code-workspace:4]: Yes, it's to access files at the root level. It excludes the `packages` directory.

**Review by louis-bompart — APPROVED** [review]
Probably won't use it to be perfectly honest, but looks cool nonetheless 👍

**Review by aprudhomme-coveo — APPROVED** [review]
Nice, I like it! I will begin using it :).

**Pull request merged**